pyfr/partitioners/base.py

```diff
@@ -241,7 +241,7 @@ def partition(self, mesh):
             newmesh = mesh
 
         # Generate a new UUID for the mesh
-        newmesh['mesh_uuid'] = newuuid = str(uuid.uuid4())
+        newmesh['mesh_uuid'] = newuuid = uuid.uuid4().hex.encode()
```

Comment:
Not entirely sure what is going on here.  The UUID is now a bytes instead of a string.  Thus far everything has in the file format has either been an array (float/double) or a string (which we sometimes need to kick a bit).